### ISSUE TITLE
[Bug-Candidate]: For Unary IR operations, the `type` field is a UnaryOperationType instead of UnaryType

### ISSUE THREAD

**Issue by kevinclancy** [crytic/slither#2085]
### Describe the issue:

In Unary IR operations, the `type` field records what type of unary operation is performed. I expect this field to have type `UnaryType`, much like the type field of `Binary` has type `BinaryType`, but instead it has type `UnaryOperationType`.

`UnaryOperationType` seems inappropriate for this field, because it is an enum containing variants like `DELETE` and `PLUSPLUS_PRE`, which never actually occur in IR.

### Code example to reproduce the issue:

contract Test {
    function test(uint64 a) internal pure returns (uint32) {
        require(!(a > 10));
        return uint32(a); 
    }
}

### Version:

0.9.6

### Relevant log output:

```shell
Set breakpoints in the `__str__` methods for `Unary` and `Binary` and examine the types of the the `type` field.
```

**Comment by 0xalpharush** [comment]
Yeah it should be converted to UnaryType here https://github.com/crytic/slither/blob/e5f2a86f0906fd62c6c4eccb9dbfa5ab30671a78/slither/visitors/slithir/expression_to_slithir.py#L586-L591
as is done for Binary here
https://github.com/crytic/slither/blob/e5f2a86f0906fd62c6c4eccb9dbfa5ab30671a78/slither/visitors/slithir/expression_to_slithir.py#L277
Also need to update the types to `UnaryType`
https://github.com/crytic/slither/blob/e5f2a86f0906fd62c6c4eccb9dbfa5ab30671a78/slither/slithir/operations/unary.py#L38
https://github.com/crytic/slither/blob/e5f2a86f0906fd62c6c4eccb9dbfa5ab30671a78/slither/slithir/operations/unary.py#L56
